src/CalcManager/UnitConverter.cpp

```diff
@@ -192,41 +191,51 @@ void UnitConverter::SwitchActive(const wstring& newValue)
     }
 }
 
-wstring UnitConverter::CategoryToString(const Category& c, const wchar_t* delimiter)
+wstring UnitConverter::CategoryToString(const Category& c, wstring_view delimiter)
 {
-    wstringstream out(wstringstream::out);
-    out << Quote(std::to_wstring(c.id)) << delimiter << Quote(std::to_wstring(c.supportsNegative)) << delimiter << Quote(c.name) << delimiter;
-    return out.str();
+    return Quote(std::to_wstring(c.id))
+        .append(delimiter)
+        .append(Quote(std::to_wstring(c.supportsNegative)))
+        .append(delimiter)
+        .append(Quote(c.name))
+        .append(delimiter);
 }
 
-vector<wstring> UnitConverter::StringToVector(const wstring& w, const wchar_t* delimiter, bool addRemainder)
+vector<wstring> UnitConverter::StringToVector(wstring_view w, wstring_view delimiter, bool addRemainder)
 {
     size_t delimiterIndex = w.find(delimiter);
     size_t startIndex = 0;
-    vector<wstring> serializedTokens = vector<wstring>();
-    while (delimiterIndex != wstring::npos)
+    vector<wstring> serializedTokens;
+    while (delimiterIndex != wstring_view::npos)
     {
-        serializedTokens.push_back(w.substr(startIndex, delimiterIndex - startIndex));
-        startIndex = delimiterIndex + (int)wcslen(delimiter);
+        serializedTokens.emplace_back(w.substr(startIndex, delimiterIndex - startIndex));
```

Comment:
`substr` on a `wstring_view` is very cheap and returns another `wstring_view`.  This means that a `wstring` will need to be constructed for `serializedTokens` so `emplace_back` is now appropriate so the `wstring` will be constructed directly in the container.

Reply:
👍